### PR TITLE
feat(conditions): Add "flagship planet attribute" general condition

### PR DESCRIPTION
**Pull request by rovermicrover**
This will allow conditions outside of LocationFilter based on planet attributes.

For https://github.com/endless-sky/endless-sky/issues/7656
## Feature Details
This will allow conditions outside of LocationFilter based on planet attributes.

## Usage Examples
```
has "flagship planet attribute: temperate climate"
```

### Testing Done & Automated Tests Added
Added an integration test 

## Performance Impact
Looking for a single membership in a set, so very low complexity.

### CODE REVIEW

File: tests/integration/config/plugins/integration-tests/data/tests/tests_flagship_planet_attribute.txt [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+			history
+		visited "Terra Incognita"
+		"visited planet" Ruin
+		conditions
+			# to skip the landing on ruin conversation
+			"Ruin: Landing: offered"
+		changes
+			planet Ruin
+				add shipyard "Remnant"
+				add outfitter "Remnant Core"
```
Comment (Hurleveur) [inline tests/integration/config/plugins/integration-tests/data/tests/tests_flagship_planet_attribute.txt:59]:
```suggestion
```
you copy pasted too quickly

**Review by Hurleveur — CHANGES_REQUESTED** [review]
looks good just is this one small thing

**Review by petervdmeer — APPROVED** [review]

File: source/PlayerInfo.cpp [hunk 2]
```diff
@@ -3110,6 +3110,17 @@ void PlayerInfo::RegisterDerivedConditions()
 	flagshipPlanetProvider.SetHasFunction(flagshipPlanetFun);
 	flagshipPlanetProvider.SetGetFunction(flagshipPlanetFun);
 
+	auto &&flagshipPlanetAttributesProvider = conditions.GetProviderPrefixed("flagship planet attribute: ");
+	auto flagshipPlanetAttributesFun = [this](const string &name) -> bool
+	{
+		if(!flagship || !flagship->GetPlanet())
+			return false;
+		string attribute = name.substr(strlen("flagship planet attribute: "));
+		return !!flagship->GetPlanet()->Attributes().count(attribute);
```
Comment (Hurleveur) [inline source/PlayerInfo.cpp:3119]: dont double ! cancel each others?

File: tests/integration/config/plugins/integration-tests/data/tests/tests_flagship_planet_attribute.txt [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+		inject "flagship planet attribute missions"
+		call "Load First Savegame"
+		apply
+			"%TEST%: FLAGSHIP PLANET ATTRIBUTE MISSION OFFER" = 1
+		call "Trigger Flagship Planet Attribute Mission"
+		call "End Flagship Planet Attribute Missions"
+		call "Depart"
+		call "Test Shipyard And Outfitter Mission"
+		assert
+			"succeed test" == 1
```
Comment (Hurleveur) [inline tests/integration/config/plugins/integration-tests/data/tests/tests_flagship_planet_attribute.txt:122]: this is a bit weird when we could just test for `"flagship planet attribute: Temperate Climate"` directly

**Review by Hurleveur — CHANGES_REQUESTED** [review]

File: source/PlayerInfo.cpp (hunk 2 above)
Comment (Hurleveur) [reply source/PlayerInfo.cpp:3119]: I'm confused because you can just not do anything and return flagship->GetPlanet()->Attributes().count(attribute);

Comment (rovermicrover) [reply source/PlayerInfo.cpp:3119]: Ah c++ does implicit conversion of int to bool so your right I can just return the count call.

File: tests/integration/config/plugins/integration-tests/data/tests/tests_flagship_planet_attribute.txt (hunk 3 above)
Comment (rovermicrover) [reply tests/integration/config/plugins/integration-tests/data/tests/tests_flagship_planet_attribute.txt:122]: I still want to test via a mission so it’s a full integration test but can definitely add these other asserts.

Comment (rovermicrover) [reply tests/integration/config/plugins/integration-tests/data/tests/tests_flagship_planet_attribute.txt:125]: Add a true assert and a false assert. Kept the mission structure for a full end to end test, but can remove that part if ya'll would prefer brevity here.

Comment (Hurleveur) [reply tests/integration/config/plugins/integration-tests/data/tests/tests_flagship_planet_attribute.txt:125]: Idk, testing via a mission doesnt seem very useful and requires one more landing depart, making the test take twice as long so I'd be for brevity.
It's not use to test if the condition is accessible by missions in this test, if we do a test for that it should be bigger and take all of them into account, this one should just focus on flagship planet attribute being set

Comment (rovermicrover) [reply tests/integration/config/plugins/integration-tests/data/tests/tests_flagship_planet_attribute.txt]: Trimmed it down, I think that works

**Review by Hurleveur — APPROVED** [review]
KISS, just how I like it. Simpler and better this way if you ask me.

**Pull request merged**